sdk/core/azure-core/inc/azure/core/internal/http/user_agent.hpp

```diff
@@ -10,26 +10,35 @@
 
 #include <string>
 
-namespace Azure { namespace Core { namespace Http { namespace _detail {
-  // NOTE: Treat Azure::Core::Http::_detail::UserAgentGenerator::GenerateUserAgent() as _internal -
-  // it is being/has been used by eventhubs.
+namespace Azure { namespace Core { namespace Http { namespace _internal {
+  /**
+   * @brief Telemetry User-Agent string generator.
+   *
+   */
   class UserAgentGenerator {
   public:
+    /**
+     * @brief Generates User-Agent string for telemetry.
+     *
+     * @param componentName the name of the SDK component.
+     * @param componentVersion the version of the SDK component.
+     * @param applicationId user application ID
+     * @param cplusplusValue value of the `__cplusplus` macro.
+     *
+     * @return User-Agent string.
+     *
+     * @see https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy
+     *
+     * @note Values for @a cplusplusValue: `__cplusplus` when value comes from the code being built
```

Comment:
Turns out the `__cplusplus` value is fixed to `199711L` on MSVC/Windows by default, which would skew our results if left as-is. There's a compiler option we could use to address this, which is what I tried here where we build the SDK packages:
https://github.com/Azure/azure-sdk-for-cpp/pull/6148

I don't know how we solve the issue generally, especially if we are planning to keep this approach, and have compiler definitions for sdk, tests, samples.

Ideally, we'd want to have a unit test to detect this issue across platforms.

Reply:
I updated the code to fall back to use the value of `_MSVC_LANG`, if customers don't use `/Zc:__cplusplus` when compiling their code with MSVC.